src/timecode.rs

```diff
@@ -497,17 +498,122 @@ impl Timecode {
     ```rust
     # use vtc::{Timecode, rates};
     let tc = Timecode::with_frames("01:00:00:00", rates::F23_98).unwrap();
-    assert_eq!("5400+00", tc.feet_and_frames())
+    assert_eq!("5400+00", tc.feet_and_frames_35mm_4p())
     ```
     */
-    pub fn feet_and_frames(&self) -> String {
-        let result = div_mod_floor(abs(self.frames()), FRAMES_PER_FOOT);
-        let feet = result.0;
-        let frames = result.1;
+    pub fn feet_and_frames_35mm_4p(&self) -> String {
```

Comment:
I have been thinking a little more about this (partially while developing the Elixir vtc library), and the more I look at this API, the less I like it.

I'm not sure I want three separate methods dedicated to F+F, it feels a little but like polluting the namespace, and is going to take up an outsized bit of realestate when getting code completion.

Instead, I think I would prefer we have only one method, and define a format Enum like so:

```rust
#[derive(Clone, Debug, Eq, PartialEq)]
pub enum FeetFrames {
    FF35mm3Perf,
    FF35mm4Perf,
    FF16mm
}
```

Then, when calling the method:

```rust
let tc = Timecode::with_frames("01:00:00:00", rates::F23_98).unwrap();
assert_eq!("5400+00", tc.feet_and_frames(FeetFrames::FF35mm4Perf))
```

Inside the function you can just match and dispatch to a private helper. The current general information can stay in the docstring, and you can move format-specific information into the docs on the Enum.

Reply:
In 3ad36a37